microsoft/testsuites/dpdk/dpdkovs.py

```diff
@@ -0,0 +1,251 @@
+            ("2.15.0", 19),  # and so on
+            ("2.17.0", 20),
+            ("2.18.0", 21),
+        ]
+        # check is version too low
+        if self.ovs_version < "2.11.0":
+            fail(
+                "OVS < 2.10 is not supported with any supported "
+                "DPDK version as of 2/8/2022"
+            )
+        # check if version too high (pairing is unknown)
+        elif self.ovs_version > "2.18.0":
+            raise NotImplementedError(
+                (
+                    f"Not implemented OVS version {self.ovs_version}. "
+                    "Please update the version requirement table in "
+                    "dpdkovs.py:check_ovs_dpdk_compatibilty with new data"
+                    "from https://docs.openvswitch.org/en/latest/faq/releases/"
+                )
+            )
+        else:
+            # iterate confirmed supported versions
+            for version_limit in ovs_dpdk_version_pairings:
+                ovs_upper_limit, dpdk_version_requirement = version_limit
+                if self.ovs_version < ovs_upper_limit:
+                    # located the correct range, check if versions match
+                    if dpdk_version.major == dpdk_version_requirement:
+                        break
+                    else:
+                        # if they don't match, fail
+                        fail(
+                            f"OVS Version {self.ovs_version} requires DPDK "
+                            f"major version of {dpdk_version_requirement}, "
+                            f"found {dpdk_version}"
+                        )
+
+    def build_with_dpdk(self, dpdk_tool: DpdkTestpmd) -> None:
+        node = self.node
+        self._check_ovs_dpdk_compatibility(dpdk_tool)
+        make = node.tools[Make]
+        dpdk_build_dir = dpdk_tool.dpdk_path.joinpath("build")
+        add_to_env = {"DPDK_BUILD": str(dpdk_build_dir)}
+        node.execute(
+            "./boot.sh",
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                "Could not run bootstrap script from OVS"
+            ),
+            update_envs=add_to_env,
+            cwd=self.repo_dir,
+        )
+        node.execute(
+            (
+                "./configure --prefix=/usr --localstatedir=/var "
+                "--sysconfdir=/etc --with-dpdk=static"
+            ),
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                "Could not configure with configure script generated by ./boot.sh"
+            ),
+            update_envs=add_to_env,
+            cwd=self.repo_dir,
+        )
+        make.make(
+            "",
+            update_envs=add_to_env,
+            cwd=self.repo_dir,
+        )
+        make.make_install(
+            update_envs=add_to_env,
+            cwd=self.repo_dir,
+        )
+
+    def test_ovs(self, device_address: str) -> None:
```

Comment:
The test logic should be in test suite folder. The tool should focus on executing logic of the tool itself. The test suite is the glue to use tools to validate.